tests/promise/promise.js

```diff
@@ -0,0 +1,29 @@
+/**
+ * @license Copyright (c) 2003-2015, CKSource - Frederico Knabben. All rights reserved.
+ * For licensing, see LICENSE.md.
+ */
+
+/* globals describe, it, expect, CKEDITOR, window */
+
+'use strict';
+
+describe( 'Promise', function() {
```

Comment:
> Then, when it comes to the returning Promises to Mocha, this is to consider and it requires Chai as Promised. I talked with @gregpabian about this as well. Still here we want to test the basics of Promise so this kind of magix is out of scope.

See: https://github.com/mochajs/mocha/pull/329 (especially the code sample in https://github.com/mochajs/mocha/pull/329#issuecomment-4546922). You don't need anything else here - just make the promise fail:

```
return promise.then( function( value ) {
    expect( value ).to.equal( 1 );
} );
```